### DISCUSSION TITLE
付费求alist百度网盘rclone挂载webdav可上传大文件方案

### DISCUSSION THREAD

**Discussion by sanwecn** [AlistGo/alist#4859] [General]
目前上传大文件报错，小文件正常上传
```ERROR RESTY Post "https://d.pcs.baidu.com/rest/2.0/pcs/superfile2?access_token=***********=upload&partseq=1&path=%252F%25E6%258A%2596%25E9%259F%25B3%252F%25E4%25B8%25AD%25E5%259B%25BD%25E8%2588%259E%25E7%258F%258D%25E5%25A6%25AE%2520%25E6%258A%2596%25E9%259F%25B32023-07-28T10_17_58.mp4&type=tmpfile&uploadid=P1-MTAuMjI5LjE4Ni4xNzM6MTY5MDUxNzQyMToyMjQxODQxOTQwODUwMTY5ODU%3D": read tcp 172.17.0.6:33080->110.242.69.44:443: read: connection reset by peer, Attempt 1```
rclone挂载参数
```rclone mount baidu: /mnt/sda/data/baidu --cache-dir /mnt/sda/data/rclonecache --vfs-cache-mode full --header "Referer:https://pan.baidu.com/"  --header "User-Agent:pan.baidu.com" --no-modtime --multi-thread-streams 8 --vfs-read-chunk-size 256M --vfs-read-chunk-size-limit 600M --buffer-size 4096M --daemon```

**Comment by pongfcnkl** [comment]
换这个项目上传
https://github.com/qjfoidnh/BaiduPCS-Go

**Reply by sanwecn** [reply]
感谢回复
我已经测试过该项目，同样只能上传小文件上传大文件会提示失败并在网盘生成0KB文件。

**Reply by pongfcnkl** [reply]
默认是有问题，建议加个policy overwrite 参数

**Reply by pongfcnkl** [reply]
你的文件是不是有百度不支持的文件名，比如emoji 特殊符号啥的

**Reply by sanwecn** [reply]
同样文件名小文件可以上传，请问你可以通过rclone挂载webdav并上传大文件吗？

**Reply by pongfcnkl** [reply]
不建议用webdav上传大文件，老老实实的用其他方式吧

**Reply by sanwecn** [reply]
请问policy overwrite参数是在BaiduPCS-Go中使用吗？能否提供具体例子？

**Reply by pongfcnkl** [reply]
直接输入进去 比如baidupcs-go upload 本地文件夹 云端文件夹 -policy=rsync

**Reply by sanwecn** [reply]
请问可以添加你的tg吗 https://t.me/xiguadaxuegao

**Reply by pongfcnkl** [reply]
直接这里说就行了

**Reply by sanwecn** [reply]
好的，我正在上传测试非常感谢你的时间和耐心

**Reply by sanwecn** [reply]
```上传文件失败, 分片上传—合并分片文件: 遇到错误, 远端服务器返回错误, 代码: 31352, 消息: commit superfile2 failed, 重试 3/3```

**Reply by pongfcnkl** [reply]
去网页端删掉0kb文件

**Reply by sanwecn** [reply]
在百度网盘网页中新建了文件夹上传也是同样的错误，该错误在https://github.com/qjfoidnh/BaiduPCS-Go/discussions/246也有其他人遇到

**Reply by pongfcnkl** [reply]
你试试这个项目
https://github.com/houtianze/bypy

**Reply by sanwecn** [reply]
好的，我尝试安装使用一下，再次感谢。

**Reply by pongfcnkl** [reply]
你的baidupcs-go是啥版本？

**Reply by sanwecn** [reply]
该项目使用的上传域名是`c.pcs.baidu.com`我的访问速度非常慢，请问如何修改他的上传服务器地址？

**Reply by sanwecn** [reply]
`BaiduPCS-Go version v3.9.2`

**Reply by pongfcnkl** [reply]
上传速度慢是因为用了单线程，多线程还没做

**Reply by sanwecn** [reply]
#root@localhost:/mnt/sda/data/go# bypy syncup -v
Loading Hash Cache File '/root/.bypy/bypy.hashcache.json'...
Hash Cache File loaded.
Gathering local directory ...
Done
Gathering remote directory ...
Done
Comparing ...
Done
<E> [18:13:16] Error accessing 'https://c.pcs.baidu.com/rest/2.0/pcs/file'
<E> [18:13:16] Function: _upload_one_file_act
<E> [18:13:16] Website parameters: {'method': 'upload', 'path': '/apps/bypy/BaiduPCS-Go-v3.9.2-linux-amd64/README.md', 'ondup': 'overwrite'}
<E> [18:13:16] Waiting 10 seconds before retrying...
<E> [18:13:26] Request Try #2 / 5
速度太慢无法使用，如果可以修改上传服务器地址应该会好点。

**Reply by pongfcnkl** [reply]
你是国外vps吧，不能把文件下载回来在上传吗？

**Reply by pongfcnkl** [reply]
![QQ截图20230728182506](https://github.com/alist-org/alist/assets/24273319/78c37d65-3766-47f8-8492-960a003acc8e)

**Reply by sanwecn** [reply]
是的没错境外大盘鸡，想把BT下载的文件上传到百度盘再下载，因为直接访问速度很慢。

**Comment by SeanHeuc** [comment]
找到问题了，百度上传因为要使用seek，所以必须先把整个文件先接受并存储为alist 临时目录temp下的临时文件，然后再去处理，此时rclone等软件会认为数据都已经上传成功了，只是等待最后服务器返回一个成功的信号。此时如果是大文件alist是需要一定的时间才能上传完，而rclone 有默认的5分钟超时，超过5分钟就会断开连接并重新上传，所以需要在任何rclone 复制等操作时加入 --timeout 
以下为示例：
rclone sync '/mnt/aaaa' 'baidu:/bbbbb' --buffer-size=256M --use-mmap -v --transfers 1 --timeout 60m

**Reply by sanwecn** [reply]
感谢你的耐心，经过测试问题照旧，同样报错

**Reply by SeanHeuc** [reply]
那请把错误日志发上来看下，我在本地测试，目前已经成功率非常高了，大文件几乎只有百分之五不到的失败率，rclone sync也会自动重试，所以完全可用。你可能因为 国内-〉 国外alist服务器 -〉国内存储服务器，路线太长所以成功率低。你把alist日志发上来我看看

**Reply by foxxorcat** [reply]
这报错明显的连接不稳定，服务器被迫断开连接，设置超时是没多大用的。
或许这个可以改善这个问题？ #4987

**Reply by SeanHeuc** [reply]
😵‍💫 太卷了，断点续传的上传实现了的话，其他网盘工具可能真的没有多少优势了

**Reply by sanwecn** [reply]
使用[BaiduPCS-Go](https://github.com/qjfoidnh/BaiduPCS-Go)可以上传，用alist挂载阿里盘可以正常上传。

**Reply by 9k001** [reply]
感谢 ，你的参数解决了我无法上传阿里云盘的问题。

**Reply by Aixtuz** [reply]
虽然不确定是否能解决楼主的问题，但确实解决了我这大文件失败的问题。
现象为：
rclone 上传进度 100% 后，显示速度降到最低。
此时查看网络状态，会发现依然在高速上传。（默认多线程上传时没注意到，后来改单线程才发现 100% 后，上传其实还没完成）
此时就会遇到这个答案说的情况，超时后再次重新上传，100% 再次重复此问题，直至默认的3次失败停止。
